flax/linen/attention.py

```diff
@@ -76,6 +77,10 @@ def dot_product_attention_weights(
     dtype: the dtype of the computation (default: infer from inputs and params)
     precision: numerical precision of the computation see `jax.lax.Precision`
       for details.
+    mdl: if not None, the attention weights are sowed into the 'intermediates'
```

Comment:
maybe explain that this is the `module` used to sow the attention weights (if given).